### PR TITLE
[feature] driver semnidaq: add support for counter acquisition

### PR DESCRIPTION
**Pull request by pieleric**
This adds support for the counting PMTs. It goes on top of PR #2655. Only the last commit is to be reviewed.

### CODE REVIEW

**Comment by pieleric** [comment]
@tepals @FCanberk , please only look at the *last* commit.

File: src/odemis/driver/test/semnidaq_test.py [hunk 1]
```diff
@@ -636,6 +684,24 @@ def test_big_acquisition(self):
         self.assertIn(model.MD_DWELL_TIME, da.metadata)
         self.assertAlmostEqual(da.metadata[model.MD_PIXEL_SIZE], exp_pxs)
 
+    def test_acquisition_long_dt(self):
+        """
+        Test image acquisition with a long dwell time (so that AO OSR > 1)
+        :return:
```
Comment (FCanberk) [inline src/odemis/driver/test/semnidaq_test.py:690]: drop the `:return:`

**Review by FCanberk — CHANGES_REQUESTED** [review]

**Comment by pieleric** [comment]
@tepals @FCanberk, now that the SEM NI DAQ driver PR is merged, this PR is fully ready for review. I've already fixed a couple of issues.

File: src/odemis/driver/test/semnidaq_test.py [hunk 2]
```diff
@@ -550,15 +567,36 @@ def test_downsample(self):
...
+                                                                         acquired_n, osr,
+                                                                         left_buffer[:grain],
+                                                                         prev_samples_n, prev_samples_sum,
+                                                                         acc_dtype,
+                                                                         average=False)
+            acquired_n += grain
+            left_buffer = left_buffer[grain:]
+
+        self.assertEqual(samples_n, 0)
+        #numpy.testing.assert_array_equal(data, data_osr1)
```
Comment (tepals) [inline src/odemis/driver/test/semnidaq_test.py:594]: why did you comment this?

File: src/odemis/driver/semnidaq.py [hunk 3]
```diff
@@ -1326,40 +1348,156 @@ def _acquire_frames(self,
...
                               ao_task: nidaqmx.Task, do_task: nidaqmx.Task, ai_task: nidaqmx.Task,
+                              ci_tasks: List[nidaqmx.Task],
                               ao_buffer_n: int, do_buffer_n: int, ai_buffer_n: int,
                               ):
         """
         Configures the synchronization tasks for AI, AO, and DO.
         :param ao_task: The AO task object.
         :param do_task: The DO task object.
         :param ai_task: The AI task object.
+        :param ci_tasks: a list of counter input tasks.
```
Comment (FCanberk) [inline src/odemis/driver/semnidaq.py:1500]:
```suggestion
        :param ci_tasks: The list of counter input tasks.
```

not much of a change, but it is just triggering my OCD :smile:

File: src/odemis/driver/semnidaq.py [hunk 4]
```diff
@@ -1418,6 +1567,7 @@ def _configure_sync_tasks(self,
                                                sample_mode=sample_mode,
                                                samps_per_chan=do_samples_n,
                                                )
+            do_task.out_stream.output_buf_size= do_buffer_n
```
Comment (FCanberk) [inline src/odemis/driver/semnidaq.py:1570]:
```suggestion
            do_task.out_stream.output_buf_size = do_buffer_n
```

File: src/odemis/driver/test/semnidaq_test.py (hunk 2 above)
Comment (pieleric) [reply src/odemis/driver/test/semnidaq_test.py]: Cannot be easily used. => just removed it .

**Review by tepals — APPROVED** [review]

**Review by FCanberk — APPROVED** [review]

**Pull request merged**